src/redis_lock/__init__.py

```diff
@@ -39,7 +39,8 @@ class Lock(object):
     A Lock context manager implemented via redis SETNX/BLPOP.
     """
 
-    def __init__(self, redis_client, name, expire=None, id=None, auto_renewal=False):
+    def __init__(self, redis_client, name, expire=None, id=None,
+                 auto_renewal=False, timeout=None):
```

Comment:
This option cannot be present here as it cannot be used with the context manager API (as discussed in #19). If we add it here we create room for user errors, confusion and anger :)